sdk/monitor/azure-monitor-opentelemetry-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulsePingSenderTests.java

```diff
@@ -92,11 +102,122 @@ void endpointChangesWithRedirectHeaderAndGetNewPingInterval() {
+        HttpPipeline httpPipeline =
+            new HttpPipelineBuilder()
+                .httpClient(request -> Mono.just(new MockHttpResponse(request, 200, httpHeaders)))
+                .tracer(new NoopTracer())
+                .build();
+        QuickPulsePingSender quickPulsePingSender =
+            new QuickPulsePingSender(
+                httpPipeline,
+                connectionString::getLiveEndpoint,
+                connectionString::getInstrumentationKey,
+                null,
+                "instance1",
+                "machine1",
+                "qpid123",
+                "testSdkVersion",
+                quickPulseConfiguration);
+        QuickPulseHeaderInfo quickPulseHeaderInfo = quickPulsePingSender.ping(null);
+        assertThat(QuickPulseStatus.QP_IS_ON).isEqualTo(quickPulseHeaderInfo.getQuickPulseStatus());
+        assertThat("https://new.endpoint.com")
+            .isEqualTo(quickPulseHeaderInfo.getQpsServiceEndpointRedirect());
+        assertThat(quickPulseConfiguration.getEtag()).isEqualTo("2::randometag::3::");
+    }
+
+    @Test
+    void successfulPingReturnsWithEtagHeaderAndRequestedMetrics() {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("x-ms-qps-service-polling-interval-hint", "1000");
+        headers.put("x-ms-qps-service-endpoint-redirect-v2", "https://new.endpoint.com");
+        headers.put("x-ms-qps-subscribed", "true");
+        headers.put("x-ms-qps-configuration-etag", "3::randometag::4::");
+        HttpHeaders httpHeaders = new HttpHeaders(headers);
+        ConnectionString connectionString = ConnectionString.parse("InstrumentationKey=fake-ikey");
+        QuickPulseConfiguration quickPulseConfiguration = new QuickPulseConfiguration();
+
+        List<Map<String, Object>> metrics = new ArrayList<>();
+        Map<String, Object> metric1 = new HashMap<>();
+        metric1.put("Id", "my_gauge");
+        metric1.put("Aggregation", "Avg");
+        metric1.put("TelemetryType", "Metric");
+        metric1.put("Projection", "my_gauge");
+        metric1.put("BackendAggregation", "Min");
+        metric1.put("FilterGroups", new ArrayList<>());
+        Map<String, Object> metric2 = new HashMap<>();
+        metric2.put("Id", "MyFruitCounter");
+        metric2.put("Aggregation", "Sum");
+        metric2.put("TelemetryType", "Metric");
+        metric2.put("Projection", "MyFruitCounter");
+        metric2.put("BackendAggregation", "Max");
+        metric2.put("FilterGroups", new ArrayList<>());
+        metrics.add(metric1);
+        metrics.add(metric2);
+
+        Map<String, Object> metricsMap = new HashMap<>();
+        metricsMap.put("Metrics", metrics);
+        ObjectMapper objectMapper = new ObjectMapper();
+        String jsonBody;
+        try {
+            jsonBody = objectMapper.writeValueAsString(metricsMap);
+        } catch (com.fasterxml.jackson.core.JsonProcessingException e) {
+            jsonBody = "{}";
+        }
```

Comment:
There is a separate PR where how we process / create json is being changed across the entire repo. This part of the code may need some changing to not use the jackson library. @heyams can give you more context about that PR and how it impacts this one.

Reply:
@harsimar @heyams any update on this?

Reply:
I believe the relevent PR is here, though it is still actively in review: https://github.com/Azure/azure-sdk-for-java/pull/41106. I anticipate that once that other PR is merged you may need to pull those changes in & use the azure json library instead.

Reply:
yes, @navsukumar might need to rebase on top of mine. I will let you know when it's ready to do so.